cache.go

```diff
@@ -70,11 +70,13 @@ func (d *direct) fillCache(cache []*Masquerade, cacheFile string) {
 			log.Debug("Saving updated masquerade cache")
 			// Truncate cache to max length if necessary
 			if len(cache) > d.maxCacheSize {
-				truncated := make([]*Masquerade, d.maxCacheSize)
+				truncated := make([]*masquerade, d.maxCacheSize)
 				copy(truncated, cache[len(cache)-d.maxCacheSize:])
 				cache = truncated
 			}
+			d.mx.RLock()
```

Comment:
We now just use a single mutex for the whole direct struct, which we lock when marshaling the masquerades into the cache.

Reply:
Actually, always passing masquerade by value in the channels would eliminate locking at all.

Reply:
Yeah, but it adds extra copy steps.